addons/account/models/account_bank_statement.py

```diff
@@ -483,60 +272,55 @@ def button_journal_entries(self):
             }
         }
 
-    def _get_last_sequence_domain(self, relaxed=False):
-        self.ensure_one()
-        where_string = "WHERE journal_id = %(journal_id)s AND name != '/'"
-        param = {'journal_id': self.journal_id.id}
-
-        if not relaxed:
-            domain = [('journal_id', '=', self.journal_id.id), ('id', '!=', self.id or self._origin.id), ('name', '!=', False)]
-            previous_name = self.search(domain + [('date', '<', self.date)], order='date desc', limit=1).name
-            if not previous_name:
-                previous_name = self.search(domain, order='date desc', limit=1).name
-            sequence_number_reset = self._deduce_sequence_number_reset(previous_name)
-            if sequence_number_reset == 'year':
-                where_string += " AND date_trunc('year', date) = date_trunc('year', %(date)s) "
-                param['date'] = self.date
-            elif sequence_number_reset == 'month':
-                where_string += " AND date_trunc('month', date) = date_trunc('month', %(date)s) "
-                param['date'] = self.date
-        return where_string, param
-
-    def _get_starting_sequence(self):
-        self.ensure_one()
-        return "%s %s %04d/%02d/00000" % (self.journal_id.code, _('Statement'), self.date.year, self.date.month)
+    def name_get(self):
+        return [(st.id, f'{st.date or ""} {st.balance_end_real}') if st.line_ids else '/' for st in self]
+
+    def _compute_display_name(self):
+        for statement in self:
+            statement.display_name = statement.name or ''
 
 
 class AccountBankStatementLine(models.Model):
     _name = "account.bank.statement.line"
     _inherits = {'account.move': 'move_id'}
     _description = "Bank Statement Line"
-    _order = "statement_id desc, date, sequence, id desc"
+    _order = "internal_index desc"
     _check_company_auto = True
 
-    # FIXME: Fields having the same name in both tables are confusing (partner_id & state). We don't change it because:
+    # FIXME: Field having the same name in both tables are confusing (partner_id). We don't change it because:
     # - It's a mess to track/fix.
     # - Some fields here could be simplified when the onchanges will be gone in account.move.
     # Should be improved in the future.
 
     # == Business fields ==
+    def default_get(self, fields):
+        defaults = super().default_get(fields)
+        # override journal_id with the default journal from the move, which is a general journal instead of liquidity
+        if 'journal_id' in fields and 'default_journal_id' not in self.env.context:
+            defaults['journal_id'] = self._get_default_journal().id
```

Comment:
Compute it from the move using the same mechanism as on the payment (context key `is_payment`)